File: 1318. Minimum Flips to Make a OR b Equal to c/main.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        flips = 0

        while a != 0 or b != 0 or c != 0:
            x, y, z = a & 0x1, b & 0x1, c & 0x1

            if z == 0:
                flips += x + y
            else:
                if (x | y) == 0:
                    flips += 1

            a >>= 1
            b >>= 1
            c >>= 1

        return flips

File: 1318. Minimum Flips to Make a OR b Equal to c/main.cpp
using namespace std;

class Solution {
public:
    int minFlips(int a, int b, int c) {
        int flips = 0;

        while (a != 0 || b != 0 || c != 0) {
            int x = a & 0x1;
            int y = b & 0x1;
            int z = c & 0x1;

            if (x == 1 && y == 1 && z == 0) {
                flips += 2;
            } else if ((x | y) != z) {
                flips += 1;
            }

            a >>= 1;
            b >>= 1;
            c >>= 1;
        }

        return flips;
    }
};